rules/composer/src/ValueObject/ComposerModifier/AddPackageToRequireDev.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Composer\ValueObject\ComposerModifier;
+
+use Rector\Composer\Modifier\ComposerModifier;
+use Rector\Composer\Contract\ComposerModifier\ComposerModifierInterface;
+use Rector\Composer\ValueObject\Version\Version;
+
+/**
+ * Only adds package to require-dev section, if package is already in composer data, nothing happen
+ * @see \Rector\Composer\Tests\ValueObject\ComposerModifier\AddPackageToRequireDevTest
+ */
+final class AddPackageToRequireDev implements ComposerModifierInterface
+{
+    /** @var string */
+    private $packageName;
+
+    /** @var Version */
+    private $version;
+
+    /**
+     * @param string $packageName name of package (vendor/package)
+     * @param string $version target package version (1.2.3, ^1.2, ~1.2.3 etc.)
+     */
+    public function __construct(string $packageName, string $version)
+    {
+        $this->packageName = $packageName;
+        $this->version = new Version($version);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function modify(array $composerData): array
+    {
+        if (!isset($composerData[ComposerModifier::SECTION_REQUIRE][$this->packageName]) && !isset($composerData[ComposerModifier::SECTION_REQUIRE_DEV][$this->packageName])) {
+            $composerData[ComposerModifier::SECTION_REQUIRE_DEV][$this->packageName] = $this->version->getVersion();
```

Comment:
`ComposerJsonSection` should be re-used: https://github.com/symplify/symplify/blob/614d76d68a92a7382d7e87d0b9ce6be4e23d1884/packages/composer-json-manipulator/src/ValueObject/ComposerJsonSection.php#L19

Reply:
this method is completely rewritten using ComposerJson object from symplify/composer-json-manipulator package